File: uri/1 - Basics/1018/Main.cs
using System; 

class URI{
    static void Main(string [] args){
        int inteiro = Int32.Parse( System.Console.ReadLine().Trim() );
        Console.Write("{0}\n", inteiro);
        int[] banknotes = {100, 50, 20, 10, 5, 2, 1};
        int aux = inteiro;
        foreach(int div in banknotes){
            Console.Write("{0} nota(s) de R$ {1},00\n", aux/div, div );
            aux = aux % div;
        }
    }
}

File: uri/1 - Basics/1018/Main.rs
use std::io;

fn main() {
    let mut input = String::new();

    // Read input from the user
    io::stdin().read_line(&mut input).expect("Failed to read line");

    // Parse input to integer
    let inteiro: i32 = input.trim().parse().expect("Invalid input");

    // Print the original integer
    println!("{}", inteiro);

    let banknotes = [100, 50, 20, 10, 5, 2, 1];
    let mut aux = inteiro;

    for &div in banknotes.iter() {
        println!("{} nota(s) de R$ {},00", aux / div, div);
        aux = aux % div;
    }
}
